sdk/include/opentelemetry/sdk/trace/default_tracer_provider.h

```diff
@@ -16,7 +17,12 @@ namespace trace
 class DefaultTracerProvider final : public opentelemetry::trace::TracerProvider
 {
 public:
-  DefaultTracerProvider() noexcept;
+  /**
+   * Construct a new tracer provider with a given processor.
+   * @param processor The processor instance to be used for all tracers
+   * obtained from this tracer provider.
+   */
+  DefaultTracerProvider(std::unique_ptr<SpanProcessor> &&processor) noexcept;
```

Comment:
I think `TraceProvider` doesn't want to have `SpanProcessor` as the parameter?
What we want to have is a method `addSpanProcessor` per the current spec?

Reply:
I added methods `GetProcessor` and `SetProcessor`, but I still left the constructor argument in place.